sdk/nodejs/asset/archive.ts

```diff
@@ -22,14 +23,14 @@ export abstract class Archive {
      * A private field to help with RTTI that works in SxS scenarios.
      */
     // tslint:disable-next-line:variable-name
-    /* @internal */ private readonly __pulumiArchive: boolean = true;
+    /* @internal */ public readonly __pulumiArchive: boolean = true;
```

Comment:
had to make these public in order to get typechecking to work properly on the "utils.isInstance" calls. We use 'keyof' to make sure that we're passing in the name correctly, and private members don't show up in the 'keyof' shape. Note: @Internal still strips this from the .d.ts, so this has no end user impact.